tests/unit/test_baseObject.py

```diff
@@ -109,3 +110,63 @@ def test_decoratedScriptsAndGestureDictionariesIfDynamic(self):
 		obj = DynamicNVDAObjectWithDecoratedScriptAndGesturesDictionary()
 		for key in ("a", "b", "c", "d", "g", "h"):
 			self.assertIn("kb:%s" % key, obj._gestureMap)
+
+class AutoPropertyObjectWithAbstractProperty(AutoPropertyObject):
+
+	_abstract_x = True
+	def _get_x(self):
+		return True
+
+	def _set_x(self, value):
+		self._attribute = value
+
+class SubclassedAutoPropertyObjectWithAbstractProperty(AutoPropertyObjectWithAbstractProperty):
+	pass
+
+class SubclassedAutoPropertyObjectWithImplementedProperty(AutoPropertyObjectWithAbstractProperty):
+
+	def _get_x(self):

```

Comment:
I think this is part of what threw me off. The property is considered implemented if either of `_get_x` or `_set_x` are implemented? Another concern, is how does the base class declare that the intended interface is "read-only" for the property.

Reply:
> The property is considered implemented if either of _get_x or _set_x are implemented?

Yes, either _set_, _get_ or _del_.

This would be the pure pythonic way

```
class foo(object):
    __metaclass__ = ABCMeta

    @abstractproperty
    def bar(self):
        return 42

class foo2(foo):

    @property
    def bar(self):
        return super(foo2, self).bar*2
```

baseObject will behave like abstractproperty if _abstract_x is True, else it behaves like property.

> Another concern, is how does the base class declare that the intended interface is "read-only" for the property.

Could you elaborate? I assume you mean somehting like this:

```
class foo(object):
	@property
	def bar(self):
		return 42

inst=foo()
inst.bar=42*2

Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: can't set attribute
```

From the doc string of AutoPropertyObject:

> If there is a _get_x but no _set_x then setting x will override the property completely.

So indeed, this differs from how python properties behave normally.